packages/embark-code-runner/src/index.ts

```diff
@@ -52,101 +50,60 @@ export default class CodeRunner {
     });
   }
 
-  private fireEmbarkJSEvents(code: string) {
-    const regexRegister = /EmbarkJS\.(.*)\.registerProvider/gm;
-    const regexSet = /EmbarkJS\.(.*)\.setProvider/gm;
-    let matches = regexRegister.exec(code);
-    if (matches) {
-      let [, provider] = matches;
-      provider = provider.toLowerCase();
-      this.providerStates[`${provider}:${ProviderEventType.ProviderRegistered}`] = true;
-      this.events.emit(`runcode:${provider}:${ProviderEventType.ProviderRegistered}`);
-    }
-    matches = regexSet.exec(code);
-    if (matches) {
-      let [, provider] = matches;
-      provider = provider.toLowerCase();
-      this.providerStates[`${provider}:${ProviderEventType.ProviderSet}`] = true;
-      this.events.emit(`runcode:${provider}:${ProviderEventType.ProviderSet}`);
-    }
-  }
-
   private registerEvents() {
-  this.events.on("runcode:register", this.registerVar.bind(this));
-
-  this.events.on("runcode:init-console-code:updated", (code: string, cb: Callback<null>) => {
-    this.evalCode(code, (err, _result) => {
-      if (err) {
-        this.logger.error("Error running init console code: ", err.message || err);
-      }
-      this.fireEmbarkJSEvents(code);
-      cb();
-    });
-  });
+    this.events.on("runcode:register", this.registerVar.bind(this));
+  }
 
-  this.events.on("runcode:embarkjs-code:updated", (code: string, cb: Callback<any>) => {
-    this.evalCode(code, (err, _result) => {
-      if (err) {
-        this.logger.error("Error running embarkjs code: ", err.message || err);
-      }
-      this.fireEmbarkJSEvents(code);
-      cb();
+  private registerCommands() {
+    this.events.setCommandHandler("runcode:getContext", (cb) => {
+      cb(this.vm.options.sandbox);
     });
-  });
-}
+    this.events.setCommandHandler("runcode:eval", this.evalCode.bind(this));
+    this.events.setCommandHandler("runcode:embarkjs:reset", this.resetEmbarkJS.bind(this));
+    this.events.on("contractsDeployed", () => this.resetEmbarkJS.bind(this));
```

Comment:
This line breaks `embark run` as it calls `resetEmbarkJS()` eagerly.

Fixed with: https://github.com/embark-framework/embark/pull/1680/commits/4421ca1d245ed66c5669a7ccedea111e7473e20f

The commit also includes an explanation why this is.